### PR TITLE
Add the ability to use a custom transport in the Mnubo client

### PR DESCRIPTION
**Pull request by aspencolby**
Mnubo client now has a property called "CustomTransport". When this is set, the defined transport will be used for all http requests made by the client.

### CODE REVIEW

**Comment by jtheoof** [comment]
Thanks @aspencolby.

First thing we now use [conventional commit](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) as a guideline. Do you mind rewording your commits to match this convention?

image

File: mnubo/client.go [hunk 1]
```diff
@@ -293,6 +294,9 @@ func (m *Mnubo) doRequest(cr ClientRequest, response interface{}) error {
 	client := &http.Client{
 		Timeout: m.Timeout,
 	}
+	if m.CustomTransport != nil {
+		client.Transport = m.CustomTransport
+	}
```
Comment (jtheoof) [inline mnubo/client.go:299]: Would it be possible to add a test for that new client API exposure ?

File: README.md [hunk 2]
```diff
@@ -367,6 +367,10 @@ root@4d7a461e5fbc:/workspaces/smartobjects-go-client# cd mnubo/
 root@4d7a461e5fbc:/workspaces/smartobjects-go-client# /usr/local/go/bin/go test -timeout 30s
 ```
 
+## Multithreading Warning
+
+This library is not optimized or tested for multi threaded usage. The client is NOT fully thread safe.
```
Comment (jtheoof) [inline README.md:372]: I would even use `*NOT*` (with the `*`) to bold that statement in markdown.

**Comment by aspencolby** [comment]
@jtheoof Thank you for the feedback. And thanks all for the review.

- I was unaware of _conventional commit_, I will modify my commits to follow that guideline. 

- I will definitely add a test case to cover the new parameter. My mistake - that should have been my first thought.

- I'll modify the README to add extra emphasis on the NOT, as suggested.

Expect to see those changes come through tomorrow morning!

**Pull request merged**